### ISSUE TITLE
Inline-формулы на хабрахабре исчезают через несколько минут после добавления

### ISSUE THREAD

**Issue by kirillfish** [vsevolod-oparin/habratex#1]
Использую `poster -c clipboard` для преобразования ТЕХ-формулы из буфера обмена в ссылку для хабра. Получается ссылка вида:

`<img src="http://tex.s2cms.ru/svg/%5Cinline%2010%5E%7B-8%7D" alt="inline_formula">`

и в предпросмотре формула выглядит как надо. Однако через несколько минут ссылка автоматически заменяется на:

`<img src="1449841773870454731881" alt="inline_formula">`

и формула перестает отображаться. В чем тут проблема?
Несколько месяцев назад inline-формулы отображались нормально.

**Comment by vsevolod-oparin** [comment]
Хм.. очень интересно, пока я писал скрипт, хабр не делал копии svg-файлов. Ваша проблема до сих воспроизводится? Если подставить новую ссылку, она тоже превращается в битую? Мне кажется, или вы меняли текст комментария?

Возможно, что проблема лежит на стороне хабрахабра. Это он делает копии картинок и заменяет их на ссылки на свое внутреннее хранилище. Если проблема повторится, могу только посоветовать сделать png-картинки вместо svg.

В файле default.json нужно заменить texsvg на texpng. При конвертации, скрипт запросит habrasid: токен, чтобы сразу грузить картинки в хабрасторедж. Как добыть, можно посмотреть [здесь](http://habrahabr.ru/post/214347/).

**Comment by kirillfish** [comment]
Да, проблема воспроизводится. Поговорил с редактором хабра, и он сказал, что с недавнего времени файлы по всем внешним ссылкам автоматически перемещаются в Habrastorage в момент сохранения черновика. Ссылка, соответственно, автоматически меняется, но становится битой при этом. Служба поддержки долго не отвечает. 

Попробую сделать что вы посоветовали с png.

**Comment by vsevolod-oparin** [comment]
Проверил svg-файлы в своих черновиках. До сих пор ничего не перезалили. Жалко, что этот процесс у них неуправляемый.

Успехов! Сообщайте, если что-то будет непонятно.

**Comment by kirillfish** [comment]
Поменял textsvg на textpng, поменял habrasid и перезапустил init.py. Картинки все равно сохраняются в svg:

`<img align="center" src="http://tex.s2cms.ru/svg/EI_%7By%5E%2A%7D%28x%29%20%3D%20-%5Cint_%7B-%5Cinfty%7D%5E%7B%5Cinfty%7D%5Ctextrm%7Bmin%7D%28y%5E%2A-y%2C0%29p_M%28y%7Cx%29dy">`

Может, я что-то делаю неправильно?

**Comment by vsevolod-oparin** [comment]
Можете прислать default.json? Или скопировать сюда содержимое?

**Comment by vsevolod-oparin** [comment]
Нашел ошибку, нужно вставить texpng, а не tex_t_png.

**Comment by kirillfish** [comment]
`{"formulas": "texpng", "habrasid": "ouod1fiqdusffit75uvdcvevh3", "encoding": "utf-8"}`
Это default.json.

Кстати говоря, сегодня Хабр перестал автоматически перезаливать все файлы в habrastorage! Не знаю, повлияли ли на это мои письма редактору и в техподдержку, и как долго это продлится.

Пока что все работает нормально, как раньше, и ссылки на tex.s2cms
